server/src/com/thoughtworks/go/server/service/ArtifactsService.java

```diff
@@ -55,6 +55,11 @@
     @Autowired
     public ArtifactsService(JobResolverService jobResolverService, StageDao stageDao,
                             ArtifactsDirHolder artifactsDirHolder, ZipUtil zipUtil, SystemService systemService) {
+        this(jobResolverService, stageDao, artifactsDirHolder, zipUtil, systemService, new ArtifactDirectoryChooser());
```

Comment:
Did not autowire it because it in the .view. package (which is not scanned by spring).
A usage check shows that it's only used in the service, will try moving it to service package.
Unless there's some weird rails wiring which will be evident when I test the UI.